### ISSUE TITLE
Broken link to Python 3.10 on macos

### ISSUE THREAD

**Issue by SethMMorton** [pypa/cibuildwheel#902]
### Description

The [configuration file](https://github.com/pypa/cibuildwheel/blob/9ce9574d4e6fce57daeb398479d39e3a388945e1/cibuildwheel/resources/build-platforms.toml) indicates that the 3.10 Python file is at https://www.python.org/ftp/python/3.10.0/python-3.10.0-macos11.pkg, but the URL returns a 404.

I notice on https://www.python.org/ftp/python/3.10.0/ that is called https://www.python.org/ftp/python/3.10.0/python-3.10.0-macos11.pkg_superceded.

The consequence is that my build breaks and I cannot deploy :( 

### Build log

https://github.com/SethMMorton/fastnumbers/runs/4075920362?check_suite_focus=true

### CI config

https://github.com/SethMMorton/fastnumbers/actions/runs/1410787082/workflow

**Comment by joerick** [comment]
Perhaps this was a temporary problem? In the directory above I see the package at:

https://www.python.org/ftp/python/3.10.0/python-3.10.0-macos11.pkg

Could you check again on your side?

**Comment by SethMMorton** [comment]
I confirm that it appears now and the link I originally provided does not. That is *incredibly* frustrating (but not your fault!).

I'll try again, and if it works now I'll close this issue.

**Comment by arnimarj** [comment]
The current situation on https://www.python.org/ftp/python/3.10.0/ is this:

```
...
python-3.10.0post2-macos11.pkg                     03-Nov-2021 02:06            39735213
python-3.10.0post2-macos11.pkg.asc                 03-Nov-2021 02:06                 833
...
python-3.10.0-embed-win32.zip.asc                  04-Oct-2021 19:27                 836
python-3.10.0-macos11.pkg.asc_superceded           04-Oct-2021 19:13                 833
...
```

I'm not entirely sure what's going on

**Comment by astrofrog** [comment]
I'm running into this today on Azure Pipelines:

```
-11-03T13:54:41.4111440Z ##[group]Installing Python cp310...
2021-11-03T13:54:41.4111860Z 
2021-11-03T13:54:41.5189130Z + Download https://www.python.org/ftp/python/3.10.0/python-3.10.0-macos11.pkg to /tmp/Python.pkg
2021-11-03T13:54:47.6302310Z Traceback (most recent call last):
2021-11-03T13:54:47.6304310Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/bin/cibuildwheel", line 8, in <module>
2021-11-03T13:54:47.6306310Z     sys.exit(main())
2021-11-03T13:54:47.6309160Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/cibuildwheel/__main__.py", line 198, in main
2021-11-03T13:54:47.6310770Z     cibuildwheel.macos.build(options)
2021-11-03T13:54:47.6312910Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/cibuildwheel/macos.py", line 361, in build
2021-11-03T13:54:47.6313920Z     env = setup_python(
2021-11-03T13:54:47.6315120Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/cibuildwheel/macos.py", line 169, in setup_python
2021-11-03T13:54:47.6315990Z     installation_bin_path = install_cpython(
2021-11-03T13:54:47.6317260Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/cibuildwheel/macos.py", line 128, in install_cpython
2021-11-03T13:54:47.6318110Z     download(url, Path("/tmp/Python.pkg"))
2021-11-03T13:54:47.6319420Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/cibuildwheel/util.py", line 192, in download
2021-11-03T13:54:47.6320400Z     response = urllib.request.urlopen(url, context=context)
2021-11-03T13:54:47.6321260Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/urllib/request.py", line 214, in urlopen
2021-11-03T13:54:47.6324680Z     return opener.open(url, data, timeout)
2021-11-03T13:54:47.6325870Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/urllib/request.py", line 523, in open
2021-11-03T13:54:47.6330380Z     response = meth(req, response)
2021-11-03T13:54:47.6331540Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/urllib/request.py", line 632, in http_response
2021-11-03T13:54:47.6333500Z     response = self.parent.error(
2021-11-03T13:54:47.6334620Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/urllib/request.py", line 561, in error
2021-11-03T13:54:47.6335890Z     return self._call_chain(*args)
2021-11-03T13:54:47.6336980Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/urllib/request.py", line 494, in _call_chain
2021-11-03T13:54:47.6338310Z     result = func(*args)
2021-11-03T13:54:47.6339470Z   File "/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/urllib/request.py", line 641, in http_error_default
2021-11-03T13:54:47.6341280Z     raise HTTPError(req.full_url, code, msg, hdrs, fp)
2021-11-03T13:54:47.6342240Z urllib.error.HTTPError: HTTP Error 404: Not Found
```

**Comment by astrofrog** [comment]
For some reason the package is called ``python-3.10.0-macos11.pkg_superceded ``?

**Comment by guillaumekln** [comment]
I'm facing the same issue. I found that the links "macOS 64-bit universal2 installer" in the [Python 3.10 download page](https://www.python.org/downloads/release/python-3100/) points to:

https://www.python.org/ftp/python/3.10.0/python-3.10.0post2-macos11.pkg

**Comment by SethMMorton** [comment]
Since it is reoccurring I will re-open.

**Comment by joerick** [comment]
It does seem like https://www.python.org/ftp/python/3.10.0/python-3.10.0post2-macos11.pkg is the new link. We can do another release with that URL. But I'm not sure why an old binary would be removed. Seems like perhaps it's a mistake. Could be worth a question to https://discuss.python.org/

**Comment by joerick** [comment]
I have created a PR with the fix #903. Will try to get a release out soon, perhaps tomorrow. 

Also posted on the Python forums - https://discuss.python.org/t/disappearing-macos-packages-on-python-org/11737 - I'm curious what's going on here!

**Comment by joerick** [comment]
Ned Deily got back to me on the thread above and has restored the original download at:

https://www.python.org/ftp/python/3.10.0/python-3.10.0-macos11.pkg

So cibuildwheel won't need an update, our latest release v2.2.2 is working again.
